### PR TITLE
Add tenant validation into auth valve and return HTML error page if content type is not application/json

### PR DESCRIPTION
**Pull request by AnuradhaSK**
### Proposed changes in this pull request

Resolves https://github.com/wso2/product-is/issues/10741

## After merging this PR

- [ ] bump identity-carbon-auth-rest version in product is
- [ ] Merge https://github.com/wso2/product-is/pull/10902

### CODE REVIEW

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java [hunk 1]
```diff
@@ -217,4 +227,74 @@ private void unsetCurrentSessionIdThreadLocal() {
...
+                        tenantDomain);
+                return false;
+            }
+        } catch (UserStoreException ex) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred while validating tenant domain.", ex);
+            }
+            handleInvalidTenantDomainErrorResponse(request, response, HttpServletResponse.SC_NOT_FOUND, tenantDomain);
+            return false;
+        } catch (IdentityRuntimeException e) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:248]: In which cases can an IdentityRuntimeException be thrown?

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java [hunk 2]
```diff
@@ -217,4 +227,74 @@ private void unsetCurrentSessionIdThreadLocal() {
...
+            String errorPage = AuthenticationValveDataHolder.getInstance().getInvalidTenantDomainErrorPage();
+            errorPage = errorPage.replace("$tenant.name", tenantDomain);
+            response.getWriter().print(errorPage);
+        }
+    }
+
+    private void handleRuntimeErrorResponse(Response response, int error, String tenantDomain) throws
+            IOException {
+
+        response.setContentType("application/json");
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:290]: Don't we need to handle html content type here as well?

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/internal/AuthenticationValveServiceComponent.java [hunk 3]
```diff
@@ -41,11 +51,33 @@
...
+            Path invalidTenantDomainHtmlResponse =
+                    Paths.get(CarbonUtils.getCarbonHome(), "repository", "resources", "identity", "pages",
+                            "invalid_tenant_domain_response.html");
+            if (!Files.exists(invalidTenantDomainHtmlResponse) ||
+                    !Files.isRegularFile(invalidTenantDomainHtmlResponse)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("invalidTenantDomainHtmlResponse is not present at: " + invalidTenantDomainHtmlResponse);
+                }
+            }
+            File file = new File(invalidTenantDomainHtmlResponse.toString());
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/internal/AuthenticationValveServiceComponent.java:73]: Won't this throw an exception if the file is not found at the expected path?

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java [hunk 4]
```diff
@@ -217,4 +227,74 @@ private void unsetCurrentSessionIdThreadLocal() {
...
+            response.setContentType("application/json");
+            JsonObject errorResponse = new JsonObject();
+            String errorMsg = "invalid tenant domain : " + tenantDomain;
+            errorResponse.addProperty("code", error);
+            errorResponse.addProperty("message", errorMsg);
+            errorResponse.addProperty("description", errorMsg);
+            response.getWriter().print(errorResponse.toString());
+        } else {
+            response.setContentType("text/html");
+            String errorPage = AuthenticationValveDataHolder.getInstance().getInvalidTenantDomainErrorPage();
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:281]: we need to handle file does not exist case as well

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java (hunk 1 above)
Comment (AnuradhaSK) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:248]: There are two cases: 

1. https://github.com/wso2/carbon-identity-framework/blob/4ed4a4366d843bf2af432e6538c3cd91031c7d32/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L265
2. https://github.com/wso2/carbon-identity-framework/blob/4ed4a4366d843bf2af432e6538c3cd91031c7d32/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L269

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java (hunk 4 above)
Comment (AnuradhaSK) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java:284]: Fix in 15e0d61

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/internal/AuthenticationValveServiceComponent.java (hunk 3 above)
Comment (AnuradhaSK) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/internal/AuthenticationValveServiceComponent.java:73]: Fixed in 15e0d61. Added a warn log.

File: components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java (hunk 2 above)
Comment (AnuradhaSK) [reply components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java]: Fixed in 15e0d61

**Review by IsuraD — APPROVED** [review]

**Pull request merged**